backends/dpdk/dpdkHelpers.cpp

```diff
@@ -681,9 +682,52 @@ bool ConvertStatementToDpdk::preorder(const IR::MethodCallStatement *s) {
     return false;
 }
 
-// TODO(hanw): TBD
-bool ConvertStatementToDpdk::preorder(const IR::SwitchStatement *) {
-    BUG("Not implemented");
+// This function assumes EliminateSwitch midend pass is applied and only handles
+// action_run in switch expression.
+bool ConvertStatementToDpdk::preorder(const IR::SwitchStatement *s) {
+    auto tc = P4::TableApplySolver::isActionRun(s->expression, refmap, typemap);
+    BUG_CHECK(tc != nullptr, "%1%: unexpected switch statement expression",
+              s->expression);
+    auto size = s->cases.size();
+    std::vector <cstring> labels;
+    cstring label;
+    bool isdefaultPresent = false;
+    cstring default_label = "";
+    auto end_label = Util::printf_format("label_endswitch%d", next_label_id++);
+    // Emit jmp on action run statements and collect labels for each case statement
+    for (unsigned i = 0; i < size; i++) {
+        auto caseLabel = s->cases.at(i);
+        if (caseLabel->label->is<IR::DefaultExpression>()) {
```

Comment:
we definitely have a pass that makes sure the default label is always last if it exists.

Reply:
Yes. I updated the code to move the last case handling outside the loop.